app/services/unpublish_service.rb

```diff
@@ -88,6 +86,21 @@ def remove_and_redirect(edition, redirect_path, explanatory_note: nil)
 
 private
 
+  def create_withdrawal(edition, public_explanation)
+    if edition.withdrawn?
```

Comment:
This is similar to the work @kevindew is doing to preserve the withdrawal time when updating the public explanation. I wonder if it would be clearer to just `dup` any previous withdrawal and update/set the public explanation.

Reply:
Yep sure, i've amended the code to reflect this suggestion.